.bazelrc

```diff
@@ -0,0 +1,78 @@
+# use a compiler name that doesn't symlink to ccache
+build --action_env=CC=clang-16 --host_action_env=CC=clang-16
+build --action_env=CXX=clang++-16 --host_action_env=CXX=clang++-16
+build --action_env=BAZEL_COMPILER=clang
+build --cxxopt=-std=c++20 --host_cxxopt=-std=c++20
+build --linkopt -fuse-ld=lld
+
+# https://github.com/bazelbuild/rules_foreign_cc/issues/1065
+# https://github.com/bazelbuild/rules_foreign_cc/issues/1186#issuecomment-2053550487
+build --host_action_env=CXXFLAGS=-Wno-int-conversion
+build --action_env=CXXFLAGS=-Wno-int-conversion
+build --host_action_env=CFLAGS=-Wno-int-conversion
+build --action_env=CFLAGS=-Wno-int-conversion
+
+common --registry=https://bcr.bazel.build
+
+# =================================
+# Sanitizer
+# =================================
+build:sanitizer --copt -fsanitize=address,undefined,vptr,function
+build:sanitizer --linkopt -fsanitize=address,undefined,vptr,function
+build:sanitizer --linkopt --rtlib=compiler-rt
+build:sanitizer --linkopt -fsanitize-link-c++-runtime
+build:sanitizer --copt -O1
+# seastar has to be run with system allocator when using sanitizers
+build:sanitizer --@seastar//:system_allocator=True
+
+# =================================
+# Security
+# =================================
+build:cfi --copt -fsanitize=cfi
+build:cfi --copt -fvisibility=hidden
+build:cfi --copt -flto
```

Comment:
@rockwotj @StephanDollberg this comment https://github.com/redpanda-data/redpanda/pull/19946#pullrequestreview-2156957699 is the only unresolved comment from the previous PR. regarding the exact set of configuration options, it's also reasonable to lump this into the todo item on the bazel wiki related to finalizing all these decisions.

Reply:
Yeah don't think this should block anything for now but in general we should probably try to get the bazel build as close to the vtools build rather than the public one.

Reply:
>as close to the vtools build rather than the public one.

💯

Reply:
It's a little strange the compile option here is in the cfi config, but the link option is in release, either way I am LGTM

Reply:
>It's a little strange the compile option here is in the cfi config,

ahh, good spot. the `build:release --copt -flto` flag is the one that makes sense to me which i added per your comment on the other PR.

this one dangling off `--config=cfi` I translated from vtools/main.cmake without any thought. i'm going to leave everything as-is, but i've added this weirdness to the todo list to resolve. will probably need to bring in whoever was working on the security flags. maybe @michael-redpanda ?